numpy/core/multiarray.py

```diff
@@ -129,6 +139,11 @@ def empty_like(prototype, dtype=None, order=None, subok=None):
            [  4.38791518e-305,  -2.00000715e+000,   4.17269252e-309]])
 
     """
+    if (shape is not None and order not in 'CFA' and
```

Comment:
I think we need to do this check on the C side.

The function here is called to find arguments to dispatch on, e.g., if you write `np.ones_like(dask_array)`. But we don't want this method to fail on duck array types like dask just because they don't define some NumPy attributes.

Reply:
Alright, I've moved the check to the C side now.